src/sentry/tagstore/snuba/backend.py

```diff
@@ -691,6 +691,24 @@ def get_tag_value_paginator_for_projects(
             if not snuba_key.startswith("tags["):
                 dataset = Dataset.Discover
 
+        # We cannot search the values of these columns like we do other columns because they are
+        # a different type, and as such, LIKE and != do not work on them. Furthermore, because the
+        # use case for these values in autosuggestion is minimal, so we choose to disable them here.
+        #
+        # event_id:     This is a FixedString which disallows us to use LIKE on it when searching,
+        #               but does work with !=. However, for consistency sake we disallow it
+        #               entirely, furthermore, suggesting an event_id is not a very useful feature
+        #               as they are not human readable.
+        # timestamp:    This is a DateTime which disallows us to use both LIKE and != on it when
+        #               searching. Suggesting a timestamp can potentially be useful but as it does
+        #               work at all, we opt to disable it here. A potential solution can be to
+        #               generate a time range to bound where they are searching. e.g. if a user
+        #               enters 2020-07 we can generate the following conditions:
+        #               >= 2020-07-01T00:00:00 AND <= 2020-07-31T23:59:59
+        # time:         This is a column computed from timestamp so it suffers the same issues
+        if snuba_key in {"event_id", "timestamp", "time"}:
```

Comment:
If the query value is `''` should we still return no results?

Reply:
To maintain a consistent behaviour, I thought it was better to return no results in all cases for these keys instead of taking it away when the query value is non empty.

Reply:
Fair point.